src/lib.rs

```diff
@@ -21,6 +21,7 @@ extern crate alloc;
 #[macro_use]
 extern crate std as alloc;
 
+pub mod doughnut;
```

Comment:
The versioned `Doughnut` type should be exported at the top level and can leave the module private.

Otherwise users will have to do:
`use doughnut_rs::doughnut::Doughnut`

```rust
mod doughnut;
pub use Doughnut;
```